src/test/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/MigrationTests.java

```diff
@@ -137,6 +137,31 @@ public void testSecurityValidateWithInvalidConfig() throws Exception {
 
     }
 
+    @Test
+    public void testSecurityMigrateWithEmptyPassword() throws Exception{
+        final Settings settings = Settings.builder().put(SSLConfigConstants.OPENDISTRO_SECURITY_SSL_HTTP_CLIENTAUTH_MODE, "REQUIRE")
+                .put("opendistro_security.ssl.http.enabled", true)
+                .put("opendistro_security.ssl.http.keystore_filepath", FileHelper.getAbsoluteFilePathFromClassPath("migration/node-0-keystore.jks"))
+                .put("opendistro_security.ssl.http.truststore_filepath", FileHelper.getAbsoluteFilePathFromClassPath("migration/truststore.jks"))
+                .put(ConfigConstants.OPENDISTRO_SECURITY_UNSUPPORTED_ACCEPT_INVALID_CONFIG, true)
+                .build();
+        setup(Settings.EMPTY, new DynamicSecurityConfig().setSecurityInternalUsers("internal_users2.yml").setLegacy(), settings, true);
+        final RestHelper rh = restHelper(); //ssl resthelper
+
+        rh.enableHTTPClientSSL = true;
+        rh.trustHTTPServerCertificate = true;
+        rh.sendAdminCertificate = true;
+        rh.keystore = "kirk-keystore.jks";
+
+        String internalUsersWithEmptyPassword = "{\"logstash\":{\"hash\":\"\",\"roles\":[\"logstash\"]},\"Stephen_123\":{\"hash\":\"\", \"password\":\"\"},\"snapshotrestore\":{\"hash\":\"\",\"roles\":[\"snapshotrestore\"]},\"admin\":{\"attributes\":{\"attribute1\":\"value1\",\"attribute3\":\"value3\",\"attribute2\":\"value2\"},\"readonly\":\"true\",\"hash\":\"\",\"roles\":[\"admin\"]},\"kibanaserver\":{\"readonly\":\"true\",\"hash\":\"\"},\"kibanaro\":{\"hash\":\"\",\"roles\":[\"kibanauser\",\"readall\"]},\"readall\":{\"hash\":\"\",\"roles\":[\"readall\"]}}";
```

Comment:
nitpick: include only whats relevant for the test

Reply:
I was thinking of that too. But I'd like to keep it consistent with what API does as much as possible. I kept the default internalusers, and just created a new internal user via API and add an empty password manually. I didn't deleted anything else.

Reply:
FYI, default users are shipped only with default configuration. They are not necessary to be there for any consistency

Reply:
Ok. Good to know that!